### PR TITLE
chore: enhance argoCD delete operation to handle exceptions better

### PR DESCRIPTION
**Pull request by sbellam1187**
#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

fixes: https://github.com/RoadieHQ/roadie-backstage-plugins/issues/846

### CODE REVIEW

**Comment by sbellam1187** [comment]
The changes part of the PR are around argocd backend plugin and haven't touched any other plugin, but some UI tests are failing.

**Comment by kissmikijr** [comment]
Retriggered the failed job, not sure what's going on.

**Review by kissmikijr — CHANGES_REQUESTED** [review]
Thank you! Just some small tweaks

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts [hunk 1]
```diff
@@ -224,72 +211,96 @@ export function createRouter({
...
       } catch (e: any) {
-        logger.error(e);
-        return response.status(500).send({
-          status:
-            typeof e.message === 'string'
-              ? e.message
-              : 'error with deleteing argo app',
-        });
+        if (typeof e.message === 'string') {
+          isAppexist = false;
```
Comment (kissmikijr) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts:243]:
```suggestion
          isAppExist = false;
```

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts [hunk 2]
```diff
@@ -224,72 +211,96 @@ export function createRouter({
...
+          countinueToDeleteProject = true;
+          argoDeleteAppResp.status = 'failed';
+          argoDeleteAppResp.message = e.message;
+        }
+      }
+      let isAppPendingDelete: boolean = false;
+      if (isAppexist) {
```
Comment (kissmikijr) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts:250]:
```suggestion
      if (isAppExist) {
```

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts [hunk 3]
```diff
@@ -224,72 +211,96 @@ export function createRouter({
...
+              token,
+            );
+            isAppPendingDelete = 'metadata' in argoApp;
+            if (!isAppPendingDelete) {
+              argoDeleteAppResp.status = 'success';
+              argoDeleteAppResp.message = 'application is deleted successfully';
+              break;
+            }
+            await timer(5000);
+          } catch (e: any) {
```
Comment (kissmikijr) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts:266]: Can we please log the original exception ?

File: .changeset/six-tips-study.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@roadiehq/backstage-plugin-argo-cd-backend': major
```
Comment (kissmikijr) [inline .changeset/six-tips-study.md:2]:
```suggestion
'@roadiehq/backstage-plugin-argo-cd-backend': minor
```
A minor version bump should be enough, what do you think? Are we introducing breaking change here?

**Review by kissmikijr — CHANGES_REQUESTED** [review]
just one more thing and ready to go!

**Comment by sbellam1187** [comment]
Merged latest changes from main branch and observing few tests are failing with the changes not part of the the PR. How to proceed with the change? @karlhaworth

**Review by kissmikijr — APPROVED** [review]
I think it is ready to merge, @karlhaworth any more thoughts on this?

**Comment by kissmikijr** [comment]
Let's ship it

**Pull request merged**